### ISSUE TITLE
Unexpected unknown pseudo-class selector ":deep" (selector-pseudo-class-no-unknown)

### ISSUE THREAD

**Issue by Chen-913** [stylelint/stylelint#6203]
### What steps are needed to reproduce the bug?

I use Vue3 and draw maps using mapboxgl, and the Stylelint prompts me for `Unexpected unknown pseudo-class selector ":deep" (selector-pseudo-class-no-unknown)` when I use Vue3's latest `.deep` for style piercing

I also used the Stylelint plugin from vscode, version `0.87.6`, but due to project constraints I had to choose this version over the latest one

### What Stylelint configuration is needed to reproduce the bug?

```js
module.exports = {
  root: true,
  extends: [
    'stylelint-config-standard',
    'stylelint-config-recess-order',
  ],
  rules: {
    indentation: 2,
    'selector-pseudo-class-no-unknown': [
      true,
      {
        ignorePseudoClasses: ['deep'],
      },
    ],
  },
};
```

### How did you run Stylelint?

stylelint

### Which version of Stylelint are you using?

13.13.1

### What did you expect to happen?

This error should not occur if I have written the relevant rules

### What actually happened?

![image](https://user-images.githubusercontent.com/72293244/179175304-272ead7c-1861-4ccc-bd8e-325c93f16d80.png)

### Does the bug relate to non-standard syntax?

Less

**Comment by Chen-913** [comment]
The project still worked and the styles worked, but my vscode editor gave me an error

**Comment by ybiquitous** [comment]
@Chen-913 Thanks for filing the issue and using the template.

I tried to make the minimum reproduction on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcDYwA4AoB0AzCECUABMAL4A6AdiADQgDOAnuQC4CGAHguLbdeBOVgCWAc07AKBAqRAAnAK4AbGLWnwiEyVLowlYJhBkBaFLRhyoEQ2AUtuh8pbnkA1g4Du5VQQDaGzQSZ5GCo-TXFyf39pEQcZGAAFU3MIAGEbbmUvXwjIqJBkFGlQ-wBdYoIyHMkynMriEGIgA), but I could not.

Could you please reproduce the error on the demo?

**Comment by Chen-913** [comment]
@ybiquitous I tried to develop on another computer, and the same version of the vscode plug-in did not report an error. I think it should be my computer.
